tests/data_access/RoomsDAOTest.java

```diff
@@ -174,4 +174,36 @@ public Room addRoom(User user, LoginUserDataAccessInterface userDao, String room
         RuntimeException.class,
         () -> dao.addRoom(dummyUser, null, roomName));
   }
+
+  @Test
+  public void changeRoomNameSuccess() {
+    OkHttpClient client = new OkHttpClient();
+    RoomsDataAccessInterface dao = new FirebaseRoomsDataAccessObject(client);
+    LoginUserDataAccessInterface userDao = new FirebaseUserDataAccessObject(client);
+    User dummyUser = addFirebaseDummyUser();
+    Room room = dao.addRoom(dummyUser, userDao, "New Room");
+    dao.changeRoomName(dummyUser, userDao, room, "Test Room 2");
+    Room retrievedRoom = dao.getRoomFromId(dummyUser, userDao, room.getUid());
+    Assert.assertEquals("Test Room 2", retrievedRoom.getName());
+    cleanUpRoom(room, dummyUser);
+    cleanUpUser(dummyUser);
+  }
+
+  @Test
+  public void changeRoomNameFailure() {
+    RoomsDataAccessInterface dao =
+        new FirebaseRoomsDataAccessObject(null) {
+          @Override
+          public void changeRoomName(
+              User user, LoginUserDataAccessInterface userDao, Room activeRoom, String roomName) {
+            throw new RuntimeException("Unable to change room name. Please try again.");
```

Comment:
You should make an actual call to Firebase to test the exact error messages that the user will encounter; you can then change this to checking a prefix + some variable Firebase error

Reply:
See the other tests in this file, which this is imitating